engine/src/operators/OrderBy.cpp

```diff
@@ -366,12 +540,13 @@ std::unique_ptr<ral::frame::BlazingTable> logicalLimit(
 			std::unique_ptr<cudf::column> output = cudf::experimental::copy_range(table.view().column(i), *mycolumn, 0, limitRows, 0);
 			output_cols.push_back(std::move(output));
 		}
-	}
-
-	return std::make_unique<ral::frame::BlazingTable>( 
-		std::make_unique<cudf::experimental::table>( std::move(output_cols) ), table.names() 
-	);
+		return std::make_unique<ral::frame::BlazingTable>( 
+			std::make_unique<cudf::experimental::table>( std::move(output_cols) ), table.names() );
+	} else {
+		return table.clone();
```

Comment:
why are you copying the data here?

Reply:
You shoiuld only have to copy in the case that you are the actual output else it shouldnt be cloning

Reply:
when we talked yesterday i explained this situation. This is an edge case. Instead of dramatically changing our standard APIs and how all this works, we would just do a clone. This is something we agreed to

Reply:
can you add a TODO o something to remind us to fix this later or make an issue?

Reply:
i will add a comment. A lot of all of this will be changing very soon, so I think just making sure that we keep this in mind for bigger than GPU is what we need